code/segmentedtickerplant/stplog.q

```diff
@@ -0,0 +1,98 @@
+// Utilites for periodic tp logging in stp process
+
+.proc.loadf[raze .proc.params`stpconfig]
+
+\d .stplg
+
+d:.z.d
+t:tables`.
+
+// Create stp log directory
+// Log structure `:stplogs/date/tabname_time
+createdld:{[name;date]
+  .os.md dldir::hsym`$getenv[`TORQHOME],(,/)"/stplogs/",string name,date
+ };
+
+// Default stp mode is tabperiod
+// TP log rolled periodically (default 1 hr), 1 log per table
+logname.tabperiod:{[dir;tab;logfreq;dailyadj]
+  ` sv(hsym dir;`$string[tab],ssr[;;""]/[-13_string logfreq xbar .z.p+dailyadj;":.D"])
+ };
+
+// Standard TP mode - to be added
+logname.none:{[dir;tab;logfreq;dailyadj]}
+
+// Custom mode, configured by user - to be added
+logname.custom:{[dir;tab;logfreq;dailyadj]}
+
+// Live logs and handles to logs for each table
+currlog:([tbl:`symbol$()]logname:`symbol$();handle:`int$())
+
+openlog:{[multilog;dir;tab;logfreq;dailyadj]
+  lname:logname[multilog][dir;tab;logfreq;dailyadj];
+  if[not type key lname;.[lname;();:;()]];
+  h:hopen lname;
+  `.stplg.currlog upsert (tab;lname;h);
+  if[addmeta;.stpm.updmeta[`open;lname;tab]];
+ };
+
+// Error log for failed updates in error mode
+openlogerr:{[dir]
+  lname:` sv(hsym dir;`$"errdatabase",string .z.d);
```

Comment:
Could we give this a better name that errdatabase, either related to the system itself (e.g. name might come from schema file) or even just "err"
Also, one error log per date seems like a good idea, but make sure it uses the correct date (based on eodtime functionality, e.g. if this was started intraday it might end up with an error log file timestamped with the wrong date)